java/java.j2seplatform/src/org/netbeans/modules/java/j2seplatform/JDKDetectorUtils.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.java.j2seplatform;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.netbeans.api.java.platform.JavaPlatform;
+import org.netbeans.api.java.platform.JavaPlatformManager;
+import org.netbeans.modules.java.j2seplatform.platformdefinition.J2SEPlatformImpl;
+import org.netbeans.spi.java.platform.JavaPlatformFactory;
+import org.openide.filesystems.FileUtil;
+import org.openide.util.Lookup;
+
+/**
+ *
+ * @author mbien
+ */
+class JDKDetectorUtils {
+
+    /**
+     * Registers new JDKs if they aren't registered yet.
+     * @param jdks List of paths to the JDKs.
+     * @param namer Converts a Path to the display name of the JDK.
+     */
+    static void registerJDKs(List<Path> jdks, Function<Path, String> namer) {
+
+        Set<String> registeredJDKNames = Stream.of(JavaPlatformManager.getDefault().getInstalledPlatforms())
+                                               .filter(JavaPlatform::isValid)
+                                               .map(JavaPlatform::getDisplayName)
```

Comment:
Theoretically asking. Shouldn't we shall use `"platform.ant.name"` property instead of DisplayName?

Reply:
what would be in that property?

a few lines later there is:
```
                        String jdkName = namer.apply(jdk);
                        if (!registeredJDKNames.contains(jdkName)) {
```
`namer.apply(jdk)` builds a display name of the JDK candidate and only registers a new one if it is not already taken.

Reply:
also worth noting: the display name would be JDK 17 (SDKMAN) or JDK 17 (System). So it is unlikely that it collides with user settings - if it does, it will simply skip the jdk

Reply:
Should possibly be aware of use of file attribute for storing registered names here?  As in https://github.com/apache/netbeans/blob/master/java/java.j2seplatform/src/org/netbeans/modules/java/j2seplatform/platformdefinition/J2SEPlatformFactory.java#L72

We could do with a better way of storing a mapping of location to display names than that mind you.

Reply:
at this point in the code, I have the path to a JDK installation folder and want to check if it is already registered. How do I generate a name from that folder and compare it with already registered JDKs, without registering it first ;)

Thats why it uses display name right now. It knows how to build that name if it registers JDKs, so it can check for JDKs which are already registered and skip the step if they are.

Having access to the path of registered JDKs would allow this too, but honestly I would probably also check the display names too since we don't want to register JDKs with the same name in the list in case of name clashes.

Reply:
How did I not see the `getInstallFolders()` method before? Anyway, changed it so that it won't register a platform if a platform with the same path or name already exists.